examples/analyze_excel_ui.py

```diff
@@ -85,50 +85,153 @@
     # Part 2: Examples on how to use processed cell sample data
     ###
 
-    # We will read IPTG concentrations for each sample from the Excel file.
-    # ``samples_table`` contains all the data from sheet "Samples", including
-    # data not directly used by ``FlowCal.excel_ui.process_samples_table()``.
-    # ``samples_table`` is a pandas dataframe, with each column having the
-    # same name as the corresponding header in the Excel file.
-    # We multiply the IPTG concentration by 1000 to convert to micromolar.
-    iptg = samples_table['IPTG (mM)']*1000
-
-    # Histogram of all samples
-    # Here, we plot the fluorescence histograms of all five samples in the same
-    # figure, using ``FlowCal.plot.hist1d``. Note how this function can be used
+    # Each entry in the Excel table has a corresponding ID, which can be used
+    # to reference information associated with or the sample loaded from a
+    # row in the Excel file. Collect the IDs of the non-control samples (i.e.,
+    # 'S0001', ..., 'S0010').
+    sample_ids = ['S00{:02}'.format(n) for n in range(1,10+1)]
+
+    # We will read DAPG concentrations from the Excel file. ``samples_table``
+    # contains all the data from sheet "Samples", including data not directly
+    # used by ``FlowCal.excel_ui.process_samples_table()``. ``samples_table``
+    # is a pandas dataframe, with each column having the same name as the
+    # corresponding header in the Excel file.
+    dapg = samples_table.loc[sample_ids,'DAPG (uM)']
+
+    # Plot 1: Histogram of all samples
+    #
+    # Here, we plot the fluorescence histograms of all ten samples in the same
+    # figure using ``FlowCal.plot.hist1d``. Note how this function can be used
     # in the context of accessory matplotlib functions to modify the axes
-    # limits and labels and add a legend, among others.
+    # limits and labels and to add a legend, among other things.
+
+    # Color each histogram according to its corresponding DAPG concentration.
+    # Use a perceptually uniform colormap (cividis), and transition among
+    # colors using a logarithmic normalization, which comports with the
+    # logarithmically spaced DAPG concentrations.
+    cmap = mpl.cm.get_cmap('cividis')
+    norm = mpl.colors.LogNorm(vmin=1e0, vmax=500)
+    colors = [cmap(norm(dapg_i)) if dapg_i > 0 else cmap(0.0)
+              for dapg_i in dapg]
```

Comment:
I would rather remove this section. I don't think it serves to illustrate anything unique about FlowCal, and adds lines to example files which are now several hundred lines long.

Reply:
Yeah, I agree the custom colormap lengthens the example. I did this because the default colormap renders a barely perceptible yellow histogram, which I think is bad data vizualization technique:

| `develop` (3affd34e00060b9212cfc9093aa9695763aa8e96) | `master` (5e187b0975499cc7887a537df81c511b0ee4798a) |
|:---:|:---:|
| <img src="https://user-images.githubusercontent.com/2179825/94490003-2f53f980-01ab-11eb-8b7a-d690070ec0e7.png" width="200" height="150"> | <img src="https://user-images.githubusercontent.com/2179825/94490116-6cb88700-01ab-11eb-849f-3314074d9088.png" width="200" height="150"> |

If we don't use a custom colormap, the current example becomes:

| `plot-violin` (5336718ae4f55574909594ccbea607483e9e7e8f) | `plot-violin` (no colormap) |
|:---:|:---:|
| <img src="https://user-images.githubusercontent.com/2179825/94490446-e2bcee00-01ab-11eb-8af6-74992f330722.png" width="200" height="150"> | <img src="https://user-images.githubusercontent.com/2179825/94490464-eea8b000-01ab-11eb-9c63-ef30fd7a6cbf.png" width="200" height="150"> |

If you think the simpler example is more important, though, I can remove the custom colormaps.

Reply:
Again, I don't feel too strongly about it. But the histograms with the default colormap look fine although not great when you open the figure. Another argument is that fixing a crappy default colormap should not be this PR's problem, so we should just use the default and not worry about it. But this is an example and which colormap we use doesn't really matter.

I don't know. I'm fine if you leave it. Seems to be the alternative that requires the least effort from where we are.

Reply:
Sounds good; I'll leave it.